lib/spack/spack/test/cmd/spec.py

```diff
@@ -50,6 +52,25 @@ def test_spec_json():
     assert 'mpich' in mpileaks
 
 
+def test_spec_format(database, config):
+    output = spec('--format', '{name}-{^mpi.name}', 'mpileaks')
+    assert output.rstrip('\n') == "mpileaks-mpich"
+
```

Comment:
Since the goal here is to test that the `--format` option works, I see no value in checking a variety of specs so recommend retaining just this one but adding `, '^mpich'` to the arguments (see below).